.github/workflows/update-cdf-library.yml

```diff
@@ -0,0 +1,144 @@
+name: Update CDF Library
+
+on:
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: write
+  pull-requests: write
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Check if the user is a maintainer
+        id: check_maintainer
+        run: |
+          USERNAME="${{ github.actor }}"
+          REPO="${{ github.repository }}"
+          IS_MAINTAINER=$(curl -s -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" "https://api.github.com/repos/$REPO/collaborators" | jq -r ".[] | select(.login == \"$USERNAME\")")
+          echo is_maintainer=$IS_MAINTAINER >> $GITHUB_OUTPUT
+      
+      - uses: actions/checkout@v3 # Checkout the repository
+        if: steps.check_maintainer.outputs.is_maintainer != '' # Only run this step if the user is a maintainer
+        
+      - name: Check for the latest version of CDF library
+        id: check_latest_version
+        if: steps.check_maintainer.outputs.is_maintainer != '' # Only run this step if the user is a maintainer
+        run: |
+          URL="https://spdf.gsfc.nasa.gov/pub/software/cdf/dist/latest/linux/"
+          FILENAME=$(curl -sL "$URL" | grep -oP 'href="[^"]*dist-cdf\.tar\.gz"' | awk -F '"' '!/\.md5$/ { print $2 }')
+
+          echo "Latest CDF version: $FILENAME"
+          echo cdf_url=$URL >> $GITHUB_OUTPUT
+          # Extract the line containing the CDF version
+          CDF_VERSION_LINE=$(grep -oP 'Version: cdf[^"]*' Dockerfile)
+
+          # Extract the version number from the line using awk and sed
+          CDF_VERSION=$(echo "$CDF_VERSION_LINE" | awk -F ': ' '{print $2}')
+
+          # Print the extracted CDF version
+          echo "CDF Version: $CDF_VERSION"
+          echo cdf_filename=$CDF_VERSION >> $GITHUB_OUTPUT
+          echo latest_version=$FILENAME >> $GITHUB_OUTPUT
+
+      - name: Download and extract CDF library
+        if: steps.check_maintainer.outputs.is_maintainer != '' # Only run this step if the user is a maintainer
+        id : download_cdf
+        run: |
+          LATEST_VERSION="${{ steps.check_latest_version.outputs.latest_version }}"
+          URL="${{ steps.check_latest_version.outputs.cdf_url }}"
+          FILENAME="${{ steps.check_latest_version.outputs.cdf_filename }}"
+          PARSED_LATEST_VERSION="${LATEST_VERSION%.tar.gz}"
+          # Check if the current version matches the latest version
+          if [ "$FILENAME" != "$PARSED_LATEST_VERSION" ]; then
```

Comment:
Should this be a `>`-like operation rather than a `!=` ? I don't see much of a difference either way.

Reply:
I might keep it at `!=` since it's the filename string instead of a specific version number